test(estree, allocator): use `default` to create `ESTreeSerializer`s

overlookmotel opened the pull request:
Pure refactor of tests related to `ESTree`.

`ESTreeSerializer` implements `Default`, so create `ESTreeSerializer`s with `default`, instead of explicitly passing `ranges: false`.

camc314 commented:
### Merge activity

* **Aug 19, 11:54 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
test(estree, allocator): use `default` to create `ESTreeSerializer`s (#13211)

Pure refactor of tests related to `ESTree`.

`ESTreeSerializer` implements `Default`, so create `ESTreeSerializer`s with `default`, instead of explicitly passing `ranges: false`.
